ADR/0027-availability-probe-framework.md

```diff
@@ -0,0 +1,68 @@
+# 27. Availability Probe Framework
+
+* Date 2023-07-06
+
+## Status
+
+Accepted
+
+## Context
+
+As a product team, we want a composite system availability metric for both Production
+and Staging environments, so that we can make roadmap decisions with easy-to-understand
+feedback about the reliability of the system in mind.
+  
+This metric should tell us the overall up time of RHTAP with respect to services under
+the control of Red Hat.
+
+RHTAP is defined to be available at a given moment if all of its components are
+reporting to be available at that given moment.
```

Comment:
If we want to visualize this metric as a composition of it's services, we might want to create something similar to this page:
https://www.githubstatus.com

Reply:
I had this one in mind as well. I don't think we should commit to it at this stage.

Reply:
Probably not. It's just a thought I wanted to share.